packages/plugin-express/src/request-info.js

```diff
@@ -5,7 +5,9 @@ module.exports = req => {
   const address = connection && connection.address && connection.address()
   const portNumber = address && address.port
   const port = (!portNumber || portNumber === 80 || portNumber === 443) ? '' : `:${portNumber}`
-  const url = `${req.protocol}://${req.hostname || req.host}${port}${req.url}`
+  const protocol = typeof req.protocol !== 'undefined' ? req.protocol : (req.connection.encrypted ? 'https' : 'http')
+  const hostname = (req.hostname || req.host || req.headers.host || '').replace(/:\d+$/, '')
```

Comment:
does req.headers definitely exist?

Reply:
yeah it's defined on the node incoming http request object: https://nodejs.org/docs/latest/api/http.html#http_message_headers